### ISSUE TITLE
No subcommand completions prompted for `pyenv` installed by the package manager on Manjaro

### ISSUE THREAD

**Issue by cantoraz** [sorin-ionescu/prezto#1886]
### Description
No subcommand completions prompted for `pyenv` installed by the package manager on Manjaro.

### Expected behavior

`pyenv` subcommand should be displayed after `pyenv<SPACE><TAB>`.

### Actual behavior

No subcommand completions, but only files under PWD displayed.

### Steps to Reproduce

1. Install the `pyenv` by using Pacman from the community repo.
2. Install Prezto, and enable the `python` module.
3. Open a terminal, type `pyenv`, `<SPACE>`, then `<TAB>`.

The `pyenv` from repo packages the completions files under `/usr/share/zsh/site-functions/`:
```zsh
$ pacman -Ql pyenv | grep zsh
pyenv /usr/share/zsh/
pyenv /usr/share/zsh/site-functions/
pyenv /usr/share/zsh/site-functions/_pyenv
```
together with completions provided by other packages:
```zsh
$ ls /usr/share/zsh/site-functions
_afew            _emulator              _kdeconnect      _protoc                    _systemd
_android         _envdir                _kernel-install  _psd                       _systemd-analyze
_archlinux-java  _exportfs              _kitchen         _pulseaudio                _systemd-delta
_artisan         _fab                   _knife           _pulseaudio-ctl            _systemd-inhibit
_atach           _fail2ban-client       _language_codes  _pyenv                     _systemd-nspawn
...
```

If I manually source `/usr/share/zsh/site-functions/_pyenv`, then subcommand completion works.
If I manually copy `/usr/share/zsh/site-functions/_pyenv` into `~/.pyenv/completions/pyenv.zsh` (just as git-cloned `pyenv`), competion doesn't work.

I've tried installing `pyenv` by git-clone under `HOME`. Completion prompt worked properly.

Subcommand completion for other commands, like `systemctl` or `openssl` works.

It seems like Prezto (or its `python` module?) didn't find or load `pyenv`'s completion file different than `~/.pyenv/completions/pyenv.zsh`.

### Versions
  - pyenv: [1.2.22-1](https://archlinux.org/packages/community/any/pyenv/)
  - Prezto commit: 704fc46
  - ZSH version: 5.8
  - OS information: Manjaro

**Comment by indrajitr** [comment]
TL;DR: This is not a Prezto issue.
---
Pyenv completion function loading is quite different :)

It does not use  Zsh `autoload` mechanism to add `_pyenv` in the shell environment. Instead, the completion function `_pyenv` is added to the shell as part of `pyenv init`. Thus the approach ArchLinux takes will not work (regardless of whether you are using Prezto or not).

See the explanation here (https://github.com/Homebrew/homebrew-core/pull/22727#issuecomment-357463849). It is in the context of Homebrew, but the reasoning is still the same.

While it is always possible to detect ArchLinux (or any other distribution) and work around it in Prezto, I personally don't think it is a good idea to deviate from pyenv's ["recommended approach"](https://github.com/pyenv/pyenv/issues/1941#issuecomment-846547202) and add more friction.

Relevant source snippets for the curious:
- `$root` initialization: https://github.com/pyenv/pyenv/blob/ca2032452b5b75acdde4e09d2ef3fdf4b1c9309c/libexec/pyenv-init#L50
- completion loading via `print_completion`: https://github.com/pyenv/pyenv/blob/ca2032452b5b75acdde4e09d2ef3fdf4b1c9309c/libexec/pyenv-init#L66
- `print_completion` function: https://github.com/pyenv/pyenv/blob/ca2032452b5b75acdde4e09d2ef3fdf4b1c9309c/libexec/pyenv-init#L196-L205

---
I am inclined to close this as _WON'T FIX_ but will let others weigh in.

/cc: @belak, @johnpneumann, @jeffwidman.

**Comment by cantoraz** [comment]
Thank you for explaining!
